ASPECT cannot compile without WorldBuilder

With the cmake flag `-DASPECT_WITH_WORLD_BUILDER=OFF`, building throws the following error:

    In file included from ...:
    ... /aspect/source/particle/property/crystal_preferred_orientation.cc:23:10: fatal error: world_builder/grains.h: No such file or directory
      23 | #include <world_builder/grains.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
    compilation terminated.
    make[2]: *** [CMakeFiles/aspect-release.dir/Unity/unity_24_cxx.cxx.o] Error 1
    make[1]: *** [CMakeFiles/aspect-release.dir/all] Error 2
    make[1]: *** Waiting for unfinished jobs....
    [ 46%] Building CXX object CMakeFiles/aspect.dir/Unity/unity_24_cxx.cxx.o
    In file included from ...:
    ... /aspect/source/particle/property/crystal_preferred_orientation.cc:23:10: fatal error: world_builder/grains.h: No such file or directory
    23 | #include <world_builder/grains.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
    compilation terminated.

Setting `-DASPECT_WITH_WORLD_BUILDER=ON` does successfully compile. I think adding

    #ifdef ASPECT_WITH_WORLD_BUILDER
    #endif

around the includes should fix it. @MFraters I haven't checked if there are more places that need these directives.

--- a/source/particle/property/crystal_preferred_orientation.cc
+++ b/source/particle/property/crystal_preferred_orientation.cc
@@ -20,9 +20,6 @@
 
 #include <aspect/particle/property/crystal_preferred_orientation.h>
 #include <aspect/geometry_model/interface.h>
-#include <world_builder/grains.h>
-#include <world_builder/world.h>
-
 #include <aspect/citation_info.h>
 #include <aspect/utilities.h>
 

